### DISCUSSION TITLE
[API] Get components truncated | Download zip files not working | How to update base language file

### DISCUSSION THREAD

**Discussion by demonoys** [WeblateOrg/weblate#6564] [Q&A] [answered]
Hi, I'm using the weblate API to create a tool for internal translations for an app and I have some issues:

1. I'm using the API method GET /api/projects/(string:project)/components/ to get all the components in a project in order to verify if they exist or not. For a particular project I have 60 components but the result is truncated to only 20 components. Any idea why the result is truncated? I'm only interested in getting the name of the components, not the whole result. Here's the code (getting the same result in postman):

```
HttpResponseMessage response = await client.GetAsync(client.BaseAddress + "/projects/" + projectSlug + "/components/");

  if (response != null && response.IsSuccessStatusCode)
  {
      content = JsonConvert.DeserializeObject<GetListObjectResponse>(Task.Run(() => response.Content.ReadAsStringAsync()).Result);
  }
```

2. This is not in the API, but there is this button on a project to download translation files as a ZIP. I used the same call that is triggered in the browser to download the zip file from my tool, and it worked fine... until we made the server public and on HTTPS. Now when I do the call, I get the home page as a result and it seems there's a "next" link somewhere in the page, which I assumed the browser redirects to in order to download the file. Any idea how can I do this in code? Here's what I do now:
```

HTTPResponseMessage response = await client.GetAsync($"{client.BaseAddress}/download/{project}/?format=zip/");

if (response != null && response.IsSuccessStatusCode)
{
    using (var fs = new FileStream(destPath, FileMode.CreateNew))
    {
        await response.Content.CopyToAsync(fs);
    }
}
```

3. Is it possible to update the base language file (english) in order to add new terms? I tried to use POST /api/translations/(string:project)/(string:component)/(string:language)/file/ and send the english file but it doesn't seem to do anything. I assume this is just to update languages that you translate.

**Comment by nijel** [comment, marked as answer]
1. The API is paginated, see https://docs.weblate.org/en/latest/api.html#authentication-and-generic-parameters
2. See https://github.com/WeblateOrg/weblate/issues/5957
3. You probably want method=source, see https://docs.weblate.org/en/latest/api.html#post--api-translations-(string-project)-(string-component)-(string-language)-file-
